### PR TITLE
Added Up Sampling (inverse Max Pooling) Layers

### PR DESCRIPTION
**Pull request by anayebi**
I've added two new layers to perform up sampling to convolutional.py, in response to: https://groups.google.com/forum/#!topic/theano-users/wy9Fuy5pWU0

The documentation for both layers is below:

UpSample1D:

Description: This layer up samples input across one dimension (e.g. inverse MaxPooling1D)
Input shape: (nb_samples, steps, dim)
Output shape: (nb_samples, upsampled_steps, dim)
Potential use case: For stacking after a MaxPooling1D Layer

UpSample2D:

Description: This layer upsamples input across two dimensions (e.g. inverse MaxPooling2D)
Input shape: (nb_samples, stack_size, nb_row, nb_col)
Output shape: (nb_samples, stack_size, new_nb_row, new_nb_col)
Potential use case: For stacking after a MaxPooling2D Layer

### CODE REVIEW

**Comment by pranv** [comment]
:+1:

File: keras/layers/convolutional.py [hunk 1]
```diff
@@ -226,6 +241,23 @@ def get_config(self):
...
+class UpSample2D(Layer):
+    def __init__(self, upsample_size=(2, 2)):
+        super(UpSample2D,self).__init__()
+        self.input = T.tensor4()
+        self.upsample_size = upsample_size
+
+
+    def get_output(self, train):
+        X = self.get_input(train)
+        Y = theano.tensor.extra_ops.repeat(X, self.upsample_size[0], axis = -2)
```
Comment (fchollet) [inline keras/layers/convolutional.py]: Why use negative axis indexes here, since the tensor shape is completely known?

Comment (anayebi) [inline keras/layers/convolutional.py]: No particular reason, just my own style preference. Would you like me to change that?

Comment (fchollet) [inline keras/layers/convolutional.py]: I think it would make the operation clearer.

Comment (anayebi) [inline keras/layers/convolutional.py]: Ok, fixed.

**Comment by anayebi** [comment]
Negative axis indices have been removed from UpSample2D to make the operation clearer.

**Comment by fchollet** [comment]
Merged. I think this would need tests, though (let's try not to add code that does not have unit tests).

**Comment by anayebi** [comment]
Certainly. I've already tested both layers extensively.

**Comment by fchollet** [comment]
Cool, I'm sure you have. I mean, specifically, adding unit tests for these layers in `tests/auto/keras/layers`.

**Pull request merged**